test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+            it('should return first match for name predicate', function () {
+                expect(_.find.apply(_, datasets.find2())).to.eql({
+                    NAME: 'Rudi',
+                    age: 15,
+                    active: true,
+                    gender: 'M'
+                });
+            });
+        });
+    });
+
+    describe('groupBy', function () {
+        describe('groupBy function', function () {
+            var params = datasets.groupBy();
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of numbers', function () {
+                var array = params[0];
+
+                expect(array.length).to.eql(7);
+                array.forEach(function (i) {
+                    expect(typeof i).to.eql('number');
+                });
+            });
+            it('should return an object with grouped square root elements', function () {
+                expect(_.groupBy.apply(_, datasets.groupBy())).to.eql({
+                    '2': [4],
+                    '3': [9],
+                    '4': [16],
+                    '5': [25],
+                    '9': [81],
+                    '11': [121],
+                    '1.4142135623730951': [2]
+                });
+            });
+        });
+        describe('groupBy2 function', function () {
+            var params = datasets.groupBy2();
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = typeof params[1];
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql('string');
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of strings and numbers', function () {
+                var array = params[0];
+                var countString = 0;
+                var countInt = 0;
+
+                array.forEach(function (i) {
+                    if (typeof i === 'string') {
+                        countString++;
+                    } else if (typeof i === 'number') {
+                        countInt++;
+                    }
+                });
+                expect(array.length).to.eql(6);
+                expect(countInt).to.eql(3);
+                expect(countString).to.eql(3);
+            });
+            it('should group element of array by length', function () {
+                expect(_.groupBy.apply(_, datasets.groupBy2())).to.eql({
+                    3: ['xyz', 'xyz'],
+                    5: ['combi'],
+                    undefined: [2, 4, 5]
+                });
+            });
+        });
+    });
+
```

Comment:
We are missing object and array iteratee types for groupBy. Moreover, we should provide some examples with a object as collection.